src/metadata/trackmetadatataglib.cpp

```diff
@@ -148,22 +152,22 @@ inline QString toQString(const TagLib::ID3v2::Frame& frame) {
 }
 
 // Returns the first frame of an ID3v2 tag as a string.
-inline QString toQStringFirst(const TagLib::ID3v2::FrameList& frameList) {
-    if (frameList.isEmpty() || (NULL == frameList.front())) {
-        return QString();
-    } else {
-        return toQString(*frameList.front());
+QString toQStringFirstNotEmpty(
+        const TagLib::ID3v2::FrameList& frameList) {
```

Comment:
StringLists are necessary for multi-valued fields that use multiple frames 
or items with the same id or key. Multiple values in a single frame are also 
supported if the standard defines a value separator like '\0' (e.g. ID3v2.4).

Some frame types additionally support a "description" to distinguish 
multiple subtypes. This is the reason why we need to read only from comment 
frames with an empty description! Otherwise the comment frames get mixed up 
and this is what TagLib does if you simply use Tag::set/getComment(). Not 
very smart, actually.

Reading the first non-empty frame is just a little bit smarter than only 
reading always the first frame of the list. It doesn't really matter, only 
for rare cases. I apply the same strategy when writing back any data, so we 
won't overwrite any data that has not been read before. Keeping the files in 
sync with the Mixxx library is required in both cases.

On 10/20/2015 08:47 PM, Daniel Schürmann wrote:
>
> In src/metadata/trackmetadatataglib.cpp 
> <https://github.com/mixxxdj/mixxx/pull/750#discussion_r42536999>:
>
> > @@ -148,22 +152,22 @@ inline QString toQString(const TagLib::ID3v2::Frame& frame) {
> >  }
> >
> >  // Returns the first frame of an ID3v2 tag as a string.
> > -inline QString toQStringFirst(const TagLib::ID3v2::FrameList& frameList) {
> > -    if (frameList.isEmpty() || (NULL == frameList.front())) {
> > -        return QString();
> > -    } else {
> > -        return toQString(*frameList.front());
> > +QString toQStringFirstNotEmpty(
> > +        const TagLib::ID3v2::FrameList& frameList) {
>
> I think I can remember, that we had similar discussion before, but I 
> cannot remember the details.
> (Who cares, we can discuss this new, with the new data on this new PR. :-P )
>
> We have a string list, when there are more than one tags of a requested 
> type in the file, right?
>
> We should be sure about the reasons and why we skip a valid empty value.
>
> Does it happen when a file has two sets of meta-data? I can think of a 
> situation when one application writes the first set and a second 
> application writes the second. I this case, we may want to display the 
> metadata from on application only.
>
> What are your use cases? What are the realistic reasons for string-lists?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/750/files#r42536999>.
>